addons/l10n_be_hr_payroll/report/hr_individual_account_reports.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class IndividualAccountReport(models.AbstractModel):
+    _name = 'report.l10n_be_hr_payroll.report_individual_account'
+    _description = 'Individual Account Report'
+
+    def _month_to_number(self, month):
+        month_dict = dict({
+            'January': 1,
+            'February': 2,
+            'March': 3,
+            'April': 4,
+            'May': 5,
+            'June': 6,
+            'July': 7,
+            'August': 8,
+            'September': 9,
+            'October': 10,
+            'November': 11,
+            'December': 12})
+        return month_dict[month.split()[0]]
+
+    def _get_data_from_report(self, data):
+        res = dict()
+        res['year'] = data['year']
+        employee = self.env['hr.employee'].browse(data['emp'])
+        payslip_rules = self.env.ref("l10n_be_hr_payroll.hr_payroll_salary_structure_employee").get_all_rules()
+        payslip_rules += self.env.ref("hr_payroll.structure_base").get_all_rules()
+        payslip_rules = sorted(payslip_rules, key=lambda x: x[1])
+
+        data_read = self.env['hr.payslip.line'].read_group([('employee_id.id', 'in', data['emp'])], ['employee_id', 'ids:array_agg(id)'], ['employee_id', "date_from:%s" % ('month')], lazy=False)
```

Comment:
Use only one search.  

Iterate on all the payslip lines, populate the dictionary {key(employee): value(dictionary)} as discussed together.

Reply:
Use dict.fromkeys to initialize the dictionary, and do something like : 
``` python
result = dict.fromkeys(employees, dict.fromkeys(rules, {'year': 0, 'month': [0]*12, 'quarter': [0]*4}))
for line in lines:
    result[employee][rule]['month'][3] += value
```